lesson_check.py parser crash on edge syntax case

Given a lesson `parsetest.md`:

```
---
title: "Parse test"
teaching: 0
exercises: 0
questions:
- "FIXME"
objectives:
- "FIXME"
keypoints:
- "FIXME"
---

> ## Title
> > ## SubTitle
> >
> > Missing empty line below crashes parser
> > ~~~
> > <h2>Beware</h2>
> > ~~~
> > {: .source}
> >
> > ~~~
> > Dragons
> > ~~~
> > {: .source }
> {: .solution}
{: .challenge}
```

running `make lesson-check-all` results in:

```
Traceback (most recent call last):
  File "bin/lesson_check.py", line 566, in <module>
    main()
  File "bin/lesson_check.py", line 123, in main
    checker.check()
  File "bin/lesson_check.py", line 342, in check
    self.check_codeblock_classes()
  File "bin/lesson_check.py", line 393, in check_codeblock_classes
    self.reporter.check(cls in KNOWN_CODEBLOCKS or cls.startswith('language-'),
AttributeError: 'NoneType' object has no attribute 'startswith'
make: *** [Makefile:139: lesson-check-all] Error 1
```

Modifying the above to read:

```
(...)
> > Missing empty line below crashes parser
> >
> > ~~~
(...)
```
avoids the crash.

The expectation is that the lesson check script can handle incorrect input without crashing, ideally providing information on what could be wrong.

Thanks.  Tried the fix in #546 by installing the gem and adding the change the `markdown_ast.rb` but when running `make lesson-check-all` I now get the error below for both the original code and the workaround "If I delete the blank line between the } and // Show help message".  Think I added the fix correctly but happy for someone else to check.  Maybe I need to reformat things differently - just assumed code block should be flexible.

```
$ make lesson-check-all
Traceback (most recent call last):
  File "intro_nextflow/bin/lesson_check.py", line 566, in <module>
    main()
  File "/intro_nextflow/bin/lesson_check.py", line 123, in main
    checker.check()
  File "/intro_nextflow/bin/lesson_check.py", line 342, in check
    self.check_codeblock_classes()
  File "/intro_nextflow/bin/lesson_check.py", line 393, in check_codeblock_classes
    self.reporter.check(cls in KNOWN_CODEBLOCKS or cls.startswith('language-'),
AttributeError: 'NoneType' object has no attribute 'startswith'
make: *** [lesson-check-all] Error 1
```

--- a/.github/workflows/template.yml
+++ b/.github/workflows/template.yml
@@ -44,7 +44,7 @@ jobs:
 
       - name: Install GitHub Pages, Bundler, and kramdown gems
         run: |
-          gem install github-pages bundler kramdown
+          gem install github-pages bundler kramdown kramdown-parser-gfm
 
       - name: Install Python modules
         run: |

--- a/.github/workflows/website.yml
+++ b/.github/workflows/website.yml
@@ -28,7 +28,7 @@ jobs:
 
       - name: Install GitHub Pages, Bundler, and kramdown gems
         run: |
-          gem install github-pages bundler kramdown
+          gem install github-pages bundler kramdown kramdown-parser-gfm
 
       - name: Install Python modules
         run: |

--- a/bin/markdown_ast.rb
+++ b/bin/markdown_ast.rb
@@ -4,9 +4,10 @@
 # Use Kramdown parser to produce AST for Markdown document.
 
 require 'kramdown'
+require 'kramdown-parser-gfm'
 require 'json'
 
 markdown = $stdin.read
-doc = Kramdown::Document.new(markdown)
+doc = Kramdown::Document.new(markdown, input: 'GFM', hard_wrap: false)
 tree = doc.to_hash_a_s_t
 puts JSON.pretty_generate(tree)

